anvil/utils.py

```diff
@@ -7,7 +7,7 @@
 from math import ceil
 import torch
 import sys
-
+import torch
```

Comment:
you haven't removed the import above. In fact the imports here are super messy, can you sort them out?

Standards then third party separated by a line and whilst we're here, may as well be alphabetical within those blocks